test/test_utils.py

```diff
@@ -33,11 +33,22 @@
 
 
 class CmdStanPathTest(unittest.TestCase):
+
     def test_default_path(self):
-        abs_rel_path = os.path.expanduser(
-            os.path.join('~', '.cmdstanpy', 'cmdstan')
-        )
-        self.assertTrue(cmdstan_path().startswith(abs_rel_path))
+        if 'CMDSTAN' in os.environ:
+            self.assertEqual(cmdstan_path(), os.environ['CMDSTAN'])
```

Comment:
will function set the env variable here and then it is taken for comparison. Should you change the order or is this wanted behaviour.

Also, should we reset the path, if it is not the default at start?

try: finally block?

Reply:
Shouldn't `CMDSTAN` be guaranteed to be set at this point (line 195 in utils.py above)?

Reply:
In other words, if it is not set, then the test should fail (or we have another test for that).

Reply:
yes, but this is a test. If user has set CMDSTAN to some specific (different than default) folder, runs tests and after tests notice that the CMDSTAN was changed.

Tests should not change anything global.

Reply:
When `cmdstan_path()` is called it sets `CMDSTAN`. So the line 39 will always be true.

Reply:
Ok I'm following it now.

Reply:
changed test so that it records value of CMDSTAN environment var before the test and uses finally block to reset accordingly.